java/piranha/src/test/java/com/uber/piranha/XPFlagCleanerTest.java

```diff
@@ -354,4 +354,177 @@ public void negative() throws IOException {
       assert_().fail("Incorrect parameters passed to the checker");
     }
   }
+
+  @Test
+  public void positiveCaseWithFlagNameAsVariable() throws IOException {
+
+    ErrorProneFlags.Builder b = ErrorProneFlags.builder();
+    b.putFlag("Piranha:FlagName", "STALE_FLAG");
+    b.putFlag("Piranha:IsTreated", "true");
+    b.putFlag("Piranha:Config", "config/piranha.properties");
+
+    try {
+      BugCheckerRefactoringTestHelper bcr =
+              BugCheckerRefactoringTestHelper.newInstance(new XPFlagCleaner(b.build()), getClass());
+
+      bcr.setArgs("-d", temporaryFolder.getRoot().getAbsolutePath());
+
+      bcr = addHelperClasses(bcr);
+      bcr.addInputLines(
+              "XPFlagCleanerSinglePositiveCase.java",
+              "package com.uber.piranha;",
+              "class XPFlagCleanerSinglePositiveCase {",
+              "private static final String STALE_FLAG_CONSTANTS = \"STALE_FLAG\";",
+              " private XPTest experimentation;",
+              " public String evaluate() {",
+              "  if (experimentation.isToggleDisabled(STALE_FLAG_CONSTANTS)) { return \"X\"; }",
+              "     else { return \"Y\";}",
+              " }",
+              "}")
+              .addOutputLines(
+                      "XPFlagCleanerSinglePositiveCase.java",
+                      "package com.uber.piranha;",
+                      "class XPFlagCleanerSinglePositiveCase {",
```

Comment:
@sivacoder The tests should fail due to the absence of Errorprone diagnostics (e.g., `// BUG: Diagnostic contains: Cleans stale XP flags",`). They are not failing locally for me. Nevertheless, can you add the diagnostics before the two cases where refactoring happens? 

@lazaroclapp Any idea why the tests aren't failing without the diagnostics? Also, can we setup travis to also run the tests?

Reply:
added